core/Symbols.cc

```diff
@@ -271,8 +272,22 @@ TypePtr ArgInfo::argumentTypeAsSeenByImplementation(Context ctx, core::TypeConst
     return Types::arrayOf(ctx, instantiated);
 }
 
-void Symbol::addMixin(const GlobalState &gs, SymbolRef sym) {
+unique_ptr<ErrorBuilder> Symbol::addMixin(const GlobalState &gs, SymbolRef sym) {
     ENFORCE(isClassOrModule());
+    // Note: Symbols without an explicit declaration may not have class or module set. They default to modules in
+    // GlobalPass.cc.
+    unique_ptr<ErrorBuilder> rv = nullptr;
+    if (sym.data(gs)->isClassModuleSet() && !sym.data(gs)->isClassOrModuleModule()) {
+        if (sym != core::Symbols::BasicObject()) {
+            if (auto e = gs.beginError(loc(), core::errors::Resolver::IncludesNonModule)) {
```

Comment:
This is the key Loc I wanted to be better. This shows the error in the file wherever the module is defined, not at each offending `include` line.

Reply:
Do you want me to pass a loc into `addMixin`?